api/src/main/java/org/openmrs/module/atomfeed/api/utils/FeedFilterUtil.java

```diff
@@ -0,0 +1,42 @@
+package org.openmrs.module.atomfeed.api.utils;
+
+import org.openmrs.PersonAddress;
+
+public final class FeedFilterUtil {
+
+	private static final String FILTER_DELIMITER = "%";
+	private static final String SKIP_VALUE = "*";
+
+	private FeedFilterUtil() { }
+
+	public static boolean isFilterValid(String filter, String filterProperty) {
+		String[] filterValues = filter.split(FILTER_DELIMITER);
+		String[] filterPropertyValues = filterProperty.split(FILTER_DELIMITER);
+
+		if (filterValues.length != filterPropertyValues.length) {
+			return false;
+		}
+
+		for (int i = 0; i < filterValues.length; i++) {
+			if (!SKIP_VALUE.equals(filterPropertyValues[i])) {
+				if (!filterPropertyValues[i].equals(filterValues[i])) {
```

Comment:
those two conditions could be wrapped in one, and probably moved to a separate method, with clear name